api/service.go

```diff
@@ -77,6 +87,114 @@ func (s *service) ModelCreate(ctx context.Context, req *pb.ModelCreateRequest) (
 	return reply, nil
 }
 
+func (s *service) ModelSave(ctx context.Context, req *pb.ModelSaveRequest) (*pb.ModelSaveReply, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method ModelSave not implemented")
+}
+
+func (s *service) ModelDelete(ctx context.Context, req *pb.ModelDeleteRequest) (*pb.ModelDeleteReply, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method ModelDelete not implemented")
+}
+
+func (s *service) ModelHas(ctx context.Context, req *pb.ModelHasRequest) (*pb.ModelHasReply, error) {
+	store, err := s.getStore(req.StoreID)
+	if err != nil {
+		return nil, err
+	}
+
+	model := store.GetModel(req.ModelName)
+	if model == nil {
+		return nil, status.Error(codes.NotFound, "model not found")
+	}
+
+	return s.processHasRequest(req, model.Has)
+}
+
+func (s *service) ModelFind(ctx context.Context, req *pb.ModelFindRequest) (*pb.ModelFindReply, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method ModelFind not implemented")
+}
+
+func (s *service) ModelFindByID(ctx context.Context, req *pb.ModelFindByIDRequest) (*pb.ModelFindByIDReply, error) {
+	store, err := s.getStore(req.StoreID)
+	if err != nil {
+		return nil, err
+	}
+
+	model := store.GetModel(req.ModelName)
+	if model == nil {
+		return nil, status.Error(codes.NotFound, "model not found")
+	}
+
+	type Foo struct{}
+
+	return s.processFindByIDRequest(req, model.FindByID, &Foo{})
```

Comment:
In this case, how do I create a default instance of the model type that is being queried to pass it into `model.FindByID`?